ddtrace/tracer.py

```diff
@@ -513,6 +517,37 @@ def _check_new_process(self):
 
         return new_ctx
 
+    def _get_queue_size(self, given_queue_size=None):
+        """Return the queue size to use for this tracer. Precedence (in order) is as follows:
+
+        1. (If defined) The given queue size
+        2. (If defined) The queue size set on the tracer
+        3. (If defined) The DD_TRACER_QUEUE_SIZE environment variable
+
+        Validation checks are performed to avoid potential misconfiguration issues.
+        """
+        if given_queue_size is not None:
+            qsize = given_queue_size
+        elif self._queue_size is not None:
+            qsize = self._queue_size
+        elif "DD_TRACER_QUEUE_SIZE" in environ:
+            qsize = int(environ.get("DD_TRACER_QUEUE_SIZE"))
```

Comment:
In profiling it's named `DD_PROFILING_MAX_EVENTS`. Maybe use something like `DD_TRACER_MAX_TRACES` or `MAX_QUEUE_SIZE`. Something that hints it's about a maximum of traces?

Reply:
"max traces" is more inline with a config option in the agent too, which is basically "max traces per second"

Reply:
I opted not to do this because I don't like "traces" as being the metric that we use here. A trace can have an arbitrary number of spans... I think spans or the byte size of the trace/spans make much more sense.

I didn't want to introduce something that I foresee as changing (possibly very soon with some of the performance issues I'm going to be tackling).

Thoughts?

Reply:
Hm, I am not sure it'd change soon. Computing the size of a trace (in terms of length or bytes) sounds costly.

And if we'd go this way, I'd postpone this PR until we actually do implement that directly, no?

Reply:
Agreed about the cost.

The reason for not postponing is that we have users actively hitting the 1k/second limit and have no way of configuring the tracer to permit more. :confused:

I didn't want to introduce too much new functionality before being able to get some concrete evidence for why it's necessary (which would take more time) 🙂